app/leaderboard/page.tsx

```diff
@@ -65,12 +63,16 @@ export default function Page() {
 
   // set user address on wallet connect and disconnect
   useEffect(() => {
+    setTimeout(() => {
+      setApiCallDelay(true);
+    }, 1000);
     if (address === "") return;
     if (address) setUserAddress(address);
     if (status === "disconnected") setUserAddress("");
   }, [address, status]);
 
   useEffect(() => {
+    if (!apiCallDelay) return;
```

Comment:
Please add a comment to explain why you are adding a 1sec delay.

Reply:
done